pkgs/stdenv/linux/default.nix

```
@@ -343,6 +342,14 @@ in
         };
       };
 
+      # force gmp to rebuild so we have the option of dynamically linking
+      # libgmp without creating a reference path from:
+      #   stage5.gcc -> stage4.coreutils -> stage3.glibc -> bootstrap
+      gmp = super.gmp.override { stdenv = self.stdenv; };
+
+      # coreutils gets rebuilt both here and also in the final stage; we rename this one to avoid confusion
+      coreutils = super.coreutils.overrideAttrs (a: a // { name = "coreutils-stage4"; });
```

The merge is unnecessary, as it's done by `overrideAttrs` already. Also `pname` not `name` should be touched here for correct results.

```suggestion
      coreutils = super.coreutils.overrideAttrs (_: { pname = "coreutils-stage4"; });
```